service/src/main/java/greencity/service/ubs/UBSManagementServiceImpl.java

```diff
@@ -1514,10 +1455,80 @@ public NotTakenOrderReasonDto getNotTakenOrderReason(Long orderId) {
             .build();
     }
 
-    @Override
-    public void saveOrderIdForRefund(Long orderId) {
-        Order order = orderRepository.findById(orderId)
-            .orElseThrow(() -> new NotFoundException(ORDER_WITH_CURRENT_ID_DOES_NOT_EXIST + orderId));
-        refundRepository.save(Refund.builder().orderId(order.getId()).build());
+    private void checkOverpayment(long overpayment) {
+        if (overpayment == 0) {
```

Comment:
can `overpayment` be negative?
Should we add a check if `overpayment` is negative?